code/Modules/Input/Core/inputMgrBase.cc

```diff
@@ -45,6 +45,12 @@ inputMgrBase::isValid() const {
 
 //------------------------------------------------------------------------------
 void
+inputMgrBase::update() {

```

Comment:
Instead of having this general update() method in the base class a better solution would be to do this locally in emscInputMgr via a PreRunLoop callback. You could have a local emscInputMgr::updateGamepads() method and add this to the PreRunLoop in emscInputMgr::setup() (similar to how the emscInputMgr::reset() method is called in the PostRunLoop).